packages/@lwc/engine/src/framework/base-lightning-element.ts

```diff
@@ -299,19 +284,16 @@ function BaseLightningElementConstructor(this: LightningElement) {
     return this as LightningElement;
 }
 
-// HTML Element - The Good Parts
 BaseLightningElementConstructor.prototype = {
     constructor: BaseLightningElementConstructor,
-    dispatchEvent(_event: Event): boolean {
+    dispatchEvent() {
         const elm = getLinkedElement(this);
         // Typescript does not like it when you treat the `arguments` object as an array
         // @ts-ignore type-mismatch;
-        return dispatchEvent.apply(elm, arguments);
+        return elm.dispatchEvent.apply(elm, arguments);
     },
     addEventListener(
-        type: string,
-        listener: EventListener,
-        options?: boolean | AddEventListenerOptions
+        ...[type, listener, options]: Parameters<typeof EventTarget.prototype.addEventListener>
```

Comment:
Parameter types didn't propagate to any of these prototype methods. I'm guessing we're running into some quirks because we're using prototypal inheritance instead of class inheritance.

Reply:
I would prefer not to rely on such syntax since the rest params syntax is carried along in the generated code, which is undesirable. I would rather prefer adding the types explicitly than having this in the generated output for performance reasons.

Reply:
That was my original concern as well but reasoned that these methods were unlikely to be called in a hot path...will revert the changes.